### ISSUE TITLE
Add cleanup(..., method="internal") to filter out functions calling .Internal()

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/globals#6]
Add `cleanup(..., method="internal")` to filter out functions calling `.Internal()`, which (1) should not be necessary to export, but more importantly (2) are not guaranteed to work consistently across R versions.

# Real-world observation
Just received the following error in a BiocParallel w/ BatchJobs run (so no more troubleshooting information):
```r
Error: 2 errors; first error:
  Error in asSamList.SamReadGroup(other): 4 arguments passed to .Internal(nchar)
 which requires 3
```
This is probably because `nchar()` was exported from an R 3.2.0 patched with:
```r
> nchar
function (x, type = "chars", allowNA = FALSE, keepNA = FALSE)
.Internal(nchar(x, type, allowNA, keepNA))
<bytecode: 0x1db8008>
<environment: namespace:base>
```
which the compute node uses an earlier version (R 3.2.0) with:
```r
> nchar
function (x, type = "chars", allowNA = FALSE)
.Internal(nchar(x, type, allowNA))
<bytecode: 0x1c8bc70>
<environment: namespace:base>
```

**Comment by jsossa117** [comment]
Hi Henrik, Have you managed to find a solution for this?

I was using a raster function and found the same error:
Error in isLonLat(x) : 
  4 arguments passed to .Internal(nchar) which requires 3
I have R 3.2.0, tracked the internal and nchar functions but cannot understand very well what the problem is.

Regards

Juan

**Comment by HenrikBengtsson** [comment]
Hi, yes, I solved this for the 'globals' package, but it sounds like your problem is irrelevant to this package.  Note that I only used the `nchar()` function as an example for this issue report.  There is a risk that this type of problem occurs whenever one stores/saves a function that uses an `.Internal()` call and later R is updated to use a modified call to `.Internal()`, e.g. `nchar()` was updated in R 3.2.0.  I recommend that you update your R versions and if your problem persists, asks for advice over on the R-help mailing list.
